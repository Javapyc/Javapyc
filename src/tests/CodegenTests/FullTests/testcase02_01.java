class Main {
	public static void main (String[] args) {
		int x	= 3;
		int y	= 4;
		int z	= x * y;
		Fact fact = new Fact();
		System.out.println(fact.factorial(z));
		System.out.println(fact.factorial2(z));
	}
}

class Fact {
	public int factorial(int i) {
		return this.factorial2(i-1);
	}
	
	public int factorial2(int i) {
		int result	= 1;
		
		while(i > 0)
		{
			result = result * i;
			i = i - 1;
		}
		
		return result;
	}
}