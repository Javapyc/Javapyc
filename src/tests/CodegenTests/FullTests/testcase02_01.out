39916800
479001600
